File: LeetcodeCore/TwoSum.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace LeetcodeCore
{
    public class TwoSumSolution
    {
        // 1. Two Sum
        public int[] TwoSum(int[] nums, int target)
        {
            var hs = new HashSet<int>();
            for (int i = 0; i < nums.Length; i++)
            {
                var j = target - nums[i];
                if (hs.Contains(j))
                    return new int[] { i, Array.IndexOf(nums, j) };
                hs.Add(nums[i]); // Add after lookup
            }
            return null;
        }
    }
}

File: LeetcodeGo/TwoSum.go
package main

// 1. Two Sum
func twoSum(nums []int, target int) []int {
	m := make(map[int]int)
	r := make([]int, 2)

	for i := 0; i < len(nums); i++ {
		m[target-nums[i]] = i
	}

	for j := 0; j < len(nums); j++ {
		val, ok := m[nums[j]]

		if ok && val != j {
			r = []int{val, j}
		}
	}

	return r
}
